packages/analyzer_buffer/pubspec.yaml

```
@@ -9,11 +9,11 @@ environment:
 resolution: workspace
 
 dependencies:
-  analyzer: ^7.3.0
+  analyzer: '>=7.3.0 <9.0.0'
   collection: ^1.19.1
   meta: ^1.15.0
   path: ^1.9.1
-  source_gen: ^3.0.0
+  source_gen: '>=3.0.0 <5.0.0'
```

```suggestion
  source_gen: '>=3.1.0 <5.0.0'
```

source_gen 4 is breaking, but this PR doesn't detect it because some transitives are preventing the bumps. To ensure compatibility, we need at least >=3.1 (w/v4 in lockfile). We'd want analyzer 8 in the lockfile to ensure everything works right.

---

analyzer_buffer does not itself use `TypeChecker.fromRuntime`

e2e does, if you uncomment it, but thats not actually blocking for this package specifically, by my understanding.

Correct me if i'm wrong.

---

Consider making a separate issue for CI stuff.
The lack of a ci test in v7 in the first place is somewhat unfortunate.

I did test v8 locally, and it works fine (except for 1 test that breaks on Windows specifically, but that's unrelated - differing slash characters)